rlclientlib/federation/vw_trainable_model.cc

```diff
@@ -33,6 +33,24 @@ void learn_and_finish_examples(VW::workspace& vw, VW::multi_ex& examples)
 
   examples.clear();
 }
+
+// Helper function to call finish_example on VW::multi_ex
+// examples is cleared at the end of this function
+void finish_examples(VW::workspace& vw, VW::multi_ex& examples)
+{
+  if (examples.empty()) { return; }
+
+  if (vw.l->is_multiline())
+  {
+    vw.finish_example(examples);
+  }
+  else
+  {
+    for (auto example : examples) { vw.finish_example(*example); }
```

Comment:
vw.finish_example(...) is overloaded for multi ex and should be called directly

Reply:
I tried it and it didn't work. Turns out `finish_example` does a check for if the learner is multiline or not.

https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/core/src/global_data.cc#L198

Reply:
That check is necessary, if you call the incorrect version based on the learner that exists it will fail.